src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -283,6 +293,46 @@ public void InitializeService(ServiceHost serviceHost, SqlToolsContext context)
 
         #region Request Handlers
 
+        /// <summary>
+        /// Completion extension load request callback
+        /// </summary>
+        /// <param name="param"></param>
+        /// <param name="requestContext"></param>
+        /// <returns></returns>
+        internal async Task HandleCompletionExtLoadRequest(CompletionExtensionParams param, RequestContext<bool> requestContext)
+        {
+            try
+            {
+                var serviceProvider = ExtensionServiceProvider.Create(new Assembly[] { AssemblyLoadContext.Default.LoadFromAssemblyPath(param.Assembly) });
+                foreach (var provider in serviceProvider.GetServices<ICompletionExtensionProvider>())
+                {
+                    var cancellationToken = new CancellationTokenSource(ExtensionLoadingTimeout).Token;
+                    var ext = await provider.CreateAsync(param.Properties ?? new Dictionary<string, object>(), cancellationToken);
+                    if (ext == null)
+                    {
+                        await requestContext.SendError("Failed to create ICompletionExtension");
+                        return;
+                    }
+                    string extName = ext.Name;
+                    await ext.Initialize(cancellationToken);
+
+                    if (!string.IsNullOrEmpty(extName))
+                    {
+                        _completionExtensions.AddOrUpdate(extName, ext, (_, previous) =>
+                        {
+                            previous?.Dispose();
+                            return ext;
+                        });
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                await requestContext.SendError(ex.Message);
+            }
```

Comment:
Don't we need to send a `requestContext.SendResult(true)` here, or `requestContext.SendResult(false)` if no provider was found? The VSCode side may be stuck waiting forever on completion of the promise.

Reply:
Fixed as you suggested